src/python/pants/backend/jvm/tasks/jvm_run.py

```diff
@@ -80,9 +80,11 @@ def execute(self):
     else:
       binary = target
 
-    # Some targets will not have extra_jvm_options in their payload,
-    # so we can't access it with target.payload.extra_jvm_options
-    extra_jvm_options = target.payload.get_field_value("extra_jvm_options")
+    # This task is installed in the "run" goal.
+    # This means that, when invoked with ./pants run, it will run regardless of whether
+    # the target is a jvm target.
+    # As a result, not all targets passed here will have good defaults for extra_jvm_options
+    extra_jvm_options = binary.payload.get_field_value("extra_jvm_options", [])
```

Comment:
Alternatively, you could move this line into the `if isinstance(.., JvmBinary)` block below?